index.js

```diff
@@ -1,76 +1,119 @@
+  })
+  .option("dry-run", {
+    type: "boolean",
+    default: false,
+    describe: "Script will not submit HTTP requests, if this flag is enabled."
+  })
+  .demandOption(["jira-url", "conf-url", "csv", "app-id", "jira-auth"])
+  .argv;
 
 const ISSUE_KEY_REGEX = /(?<=key&quot;&gt;)[A-Z]+-[0-9]+(?=&lt;)/g;
 
-// TODO Add feature for dry run - don't make HTTP requests
+const jiraEndpoint = argv['jira-url'];
+const confluenceEndpoint = argv['conf-url'];
+const csvFilename = argv['csv'];
+const appId = argv['app-id'];
+const usernameAndPassword = argv['jira-auth'];
+const isDryRun = argv['dry-run'];
 
-const jiraEndpoint = argv.jiraUrl;
-const confluenceEndpoint = argv.confUrl;
-const csvFilename = argv.csv;
-const appId = argv.appId;
+const authBuffer = Buffer.from(usernameAndPassword, 'utf-8');
+const requests = [];
+let createdLinksCount = 0;
+let preexistingLinksCount = 0;
+let errCount = 0;
 
-// Send HTTP requests to Jira
 const processRow = row => {
 
-    // Q: What about versions? Do we need to update on all versions?
+  const pageId = row["Page ID"];
+  const title = row["Page Title"];
+  const metadata = row["text_val"];
 
-    const pageId = row["Page ID"];
-    const title = row["Page Title"];
-    const metadata = row["text_val"];
+  const issueKeys = metadata.match(ISSUE_KEY_REGEX);
+  log(INFO, `[PAGE-ID: ${pageId}] Keys: ${issueKeys}`);
 
-    const issueKeys = metadata.match(ISSUE_KEY_REGEX);
-    console.log("Keys: ", issueKeys);
+  const body = {
+    globalId: `appId=${appId}&pageId=${pageId}`,
+    application: {
+      type: "com.atlassian.confluence",
+      name: "Confluence"
+    },
+    relationship: "mentioned in",
+    object: {
+      url: `${confluenceEndpoint}/pages/viewpage.action?pageId=${pageId}`,
+      title
+    }
+  };
 
-    const body = {
-        globalId: `appId=${appId}&pageId=${pageId}`,
-        application: {
-            type: "com.atlassian.confluence",
-            name: "Confluence"
-        },
-        relationship: "mentioned in",
-        object: {
-            url: `${confluenceEndpoint}/pages/viewpage.action?pageId=${pageId}`,
-            title
-        }
-    };
+  issueKeys.forEach(issueKey => {
+    const requestUrl = `${jiraEndpoint}/rest/api/latest/issue/${issueKey}/remotelink`;
 
-    console.log(body);
+    if (isDryRun) {
+      log(INFO, `[DRY-RUN] Will update ${requestUrl}`);
+      return;
+    }
 
-    issueKeys.forEach(issueKey => {
-        const requestUrl = `${jiraEndpoint}/rest/api/latest/issue/${issueKey}/remotelink`;
-        console.log(requestUrl);
-    });
+    requests.push(
+      axios.post(requestUrl, body, {
+        headers: {
+          'Authorization': `Basic ${authBuffer.toString('base64')}`,
+          'Content-Type': 'application/json'
+        }
+      })
+        .then(result => {
+          result.status === 201 ? createdLinksCount++ : preexistingLinksCount++;
```

Comment:
#ShareTheKnowledge

Working on this helped me better understand how incrementing this way in an asynchronous setting is valid in JavaScript when in Java you'd need to use an [AtomicInteger](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/atomic/AtomicInteger.html#incrementAndGet--).

This [answer](https://stackoverflow.com/a/7266985) also has a nice short summary.